sdk/core/core-client/src/serializer.ts

```diff
@@ -433,10 +433,11 @@ function serializeBasicTypes(typeName: string, objectName: string, value: any):
         !(value instanceof ArrayBuffer) &&
         !ArrayBuffer.isView(value) &&
         // File objects count as a type of Blob, so we want to use instanceof explicitly
-        !((typeof Blob === "function" || typeof Blob === "object") && value instanceof Blob)
+        !((typeof Blob === "function" || typeof Blob === "object") && value instanceof Blob) &&
+        objectType !== "function"
```

Comment:
Not sure if we can make this check more specific in a meaningful way, but am definitely open to ideas.